lib/Bio/EnsEMBL/DataCheck/Utils.pm

```diff
@@ -212,4 +212,29 @@ sub hash_diff {
   return (\%diff);
 }
 
+=item B<is_ehive_db>
+
+is_ehive_db($dba);
```

Comment:
I'm a little cautious about this change, because it could mean that failures go undetected - for example, if you run the datacheck and it passes (with some code skipped because it's in pipeline mode), the history file will record a successful run. When you subsequently remove the hive tables, and if you don't change the database name, the datacheck will skip due to the history, so the tests within the conditional would not get run. In the particular use case in this PR, I don't think this would be a problem, due to the db name change in the merging process.
But in principle, introducing different behaviour for embdded datachecks could be storing up trouble for later...
That said, I understand the reason for you introducing this function, and only the compara team create databases in this manner; so if you think it is a manageable risk, it's fine to add this method, but I would ask that you change the method name to `is_compara_ehive_db`, so that other teams know this is not something they need to concern themselves with, and do not get confused about datachecks running differently when embedded in pipelines. (An alternative approach with less potential for future harm, if you envision this sort of situation as rare, might be to have this function in the ForeignKeysCompara datacheck, rather than Utils. Or, if you do prefer not to use this method, the 'reverse-direction' FK tests could be put in a separate datacheck, which isn't included in the pipeline set.)

Reply:
I think I will change the name in the way you have suggested for now, I have another ticket regarding `ForeignKeysCompara` in which I will be able to consider the use further in another PR. Would that be okay?

Reply:
Sure, thanks Cristi.